src/test/assembly/asm/mips64-types.rs

```diff
@@ -0,0 +1,225 @@
+// no-system-llvm
+// assembly-output: emit-asm
+// compile-flags: --target mips64-unknown-linux-gnuabi64
```

Comment:
You can merge this with the MIPS32 file. See how I did it for riscv.

Reply:
I still have to duplicate some parts.
For example, mips64 use `got_disp`, mips use `got`.
mips64 uses `ld`, misp32 uses `lw` for loading addresses.

Also, llvm somehow changes $t0 to $12 in mips64.
I could circumvent it by using another register, but still.

Reply:
See [this](https://github.com/llvm/llvm-project/blob/0eab9d5823815c6520697f8d725c402c88e5d050/clang/lib/Basic/Targets/Mips.h#L370) for the register names. It seems that the new/old ABIs use different register names.

I think that we should only support plain `$<int>` register names to avoid any confusion and remove the aliases.

Reply:
yeah, that seems unfortunate. The official docs ("Introduction to the MIPS64® Architecture" and "The MIPS64® Instruction Set Reference Manual") don't talk about ABIs. A book talks about this.

Can we detect ABI and change aliases instead ?

Reply:
The macros that we use for defining the aliases don't support that. You could write a manual implementation, but it might be easier to just not support the ABI register names.

Reply:
I removed the reg aliases, and made a personal todo about adding it back.
I think although reg aliases can be confusing, people prefer to use them
because they are mnemonic.

Reply:
Merged two test files.